uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/WorkPatternTest.php

```
@@ -0,0 +1,116 @@
+use CRM_HRLeaveAndAbsences_Test_Fabricator_WorkPattern as WorkPatternFabricator;
+
+/**
+ * Class api_v3_WorkPatternTest
+ *
+ * @group headless
+ */
+class api_v3_WorkPatternTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_WorkPatternHelpersTrait;
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: contact_id, period_id
+   */
+  public function testGetCalendarRequiresContactIdAndPeriodID() {
+    civicrm_api3('WorkPattern', 'getCalendar');
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: contact_id
+   */
+  public function testGetCalendarRequiresContactIdIfPeriodIDIsNotEmpty() {
+    civicrm_api3('WorkPattern', 'getCalendar', ['period_id' => 1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: period_id
+   */
+  public function testGetCalendarRequiresPeriodIdIfContactIDIsNotEmpty() {
+    civicrm_api3('WorkPattern', 'getCalendar', ['contact_id' => 1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Unable to find a CRM_HRLeaveAndAbsences_BAO_AbsencePeriod with id 99989389121.
+   */
+  public function testGetCalendarThrowsAnErrorIfThePeriodIDIsNotForAnExistentAbsencePeriod() {
+    civicrm_api3('WorkPattern', 'getCalendar', ['contact_id' => 1, 'period_id' => 99989389121]);
+  }
+
+  public function testGetCalendarShouldReturnEmptyIfTheGivenContactDoesntExists() {
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2015-01-05'),
+      'end_date'   => CRM_Utils_Date::processDate('2015-01-19'),
+    ]);
+
+    $result = civicrm_api3('WorkPattern', 'getCalendar', ['contact_id' => 321, 'period_id' => $absencePeriod->id]);
```

Was wondering if the method should rather throw an exception or a more descriptive message if the contact does not exist rather than returning empty?